### PR TITLE
Unification of the devtools includes - 3.2

### PR DESCRIPTION
**Pull request by MichalMaler**
Aligning some snippets cosmetics for better maintenance and followup use in RHBQ 3.2.
Unifying header spaces for easier regex search.
Removing empty lines at the end of the files for better use with additional asciidoc elements such as `+`.
Fixing the style in the sentences with the xref link.

This PR is only for 3.2. No further backport needed.

### CODE REVIEW

**Review by yrodiere — COMMENTED** [review]
I don't think I'm the right person to ask for a review on this. I'll let others chime in.

**Review by jmartisk — APPROVED** [review]
Doesn't seem to actually change anything in the rendered docs, so if it helps for maintenance purposes, why not

**Comment by MichalMaler** [comment]
one change needs to be done. will set this in wip

**Comment by MichalMaler** [comment]
Setting back the WIP - Doing some more testing to be sure.

**Comment by MichalMaler** [comment]
Wanted to apply an addition of an empty  row between first and second code block for the proper asciidoc syntax highlight, but not sure if that would create some troubles down the road.

However, this format will enable us to use regexes to patch the other repo.

File: docs/src/main/asciidoc/_includes/devtools/create-cli.adoc [hunk 1]
```diff
@@ -50,13 +50,12 @@ endif::[]
 
 To create a Gradle project, add the `--gradle` or `--gradle-kotlin-dsl` option.
 
-_For more information about how to install the Quarkus CLI and use it, please refer to xref:cli-tooling.adoc[the Quarkus CLI guide]._
+For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI guide].
```
Comment (michelle-purcell) [inline docs/src/main/asciidoc/_includes/devtools/create-cli.adoc:53]:
```suggestion
For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI] guide.
```

File: docs/src/main/asciidoc/_includes/devtools/create-app.adoc [hunk 2]
```diff
@@ -52,13 +52,12 @@ ifndef::devtools-no-gradle[]
 To create a Gradle project, add the `--gradle` or `--gradle-kotlin-dsl` option.
 endif::[]
 
-_For more information about how to install the Quarkus CLI and use it, please refer to xref:cli-tooling.adoc[the Quarkus CLI guide]._
+For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI guide].
```
Comment (michelle-purcell) [inline docs/src/main/asciidoc/_includes/devtools/create-app.adoc:55]:
```suggestion
For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI] guide.
```

**Review by michelle-purcell — APPROVED** [review]
@MichalMaler - Looks great to me :+1:  This fix also helps to indirectly fix the step numbering issues with the (+) markup we've been battling with downstream also. Thank you!
:bird: :bird: :moyai:

File: docs/src/main/asciidoc/_includes/devtools/create-app.adoc [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [role="primary asciidoc-tabs-sync-cli"]
 .CLI
 ****
-[source,bash,subs=attributes+]
+[source, bash, subs=attributes+]
```
Comment (gsmet) [inline docs/src/main/asciidoc/_includes/devtools/create-app.adoc:4]: Where does this rule come from?

From what I can see, we don't put spaces here in our doc and the AsciiDoctor doc doesn't either.
So apart from requiring additional work for everyone, I don't see the value of it so an explanation would be welcome?

File: docs/src/main/asciidoc/_includes/devtools/create-app.adoc [hunk 4]
```diff
@@ -52,13 +52,12 @@ ifndef::devtools-no-gradle[]
 To create a Gradle project, add the `--gradle` or `--gradle-kotlin-dsl` option.
 endif::[]
 
-_For more information about how to install the Quarkus CLI and use it, please refer to xref:cli-tooling.adoc[the Quarkus CLI guide]._
+For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI] guide.
 ****
-
```
Comment (gsmet) [inline docs/src/main/asciidoc/_includes/devtools/create-app.adoc:57]: I'm not sure this is a good idea at all. I would rather keep the blank line.

Why did you remove it?

File: docs/src/main/asciidoc/_includes/devtools/create-cli.adoc [hunk 5]
```diff
@@ -50,13 +50,12 @@ endif::[]
 
 To create a Gradle project, add the `--gradle` or `--gradle-kotlin-dsl` option.
 
-_For more information about how to install the Quarkus CLI and use it, please refer to xref:cli-tooling.adoc[the Quarkus CLI guide]._
+For more information about how to install and use the Quarkus CLI, see the xref:cli-tooling.adoc[Quarkus CLI] guide.
 ****
-
```
Comment (gsmet) [inline docs/src/main/asciidoc/_includes/devtools/create-cli.adoc:55]: Same here.

**Review by gsmet — CHANGES_REQUESTED** [review]
I added some comments.

File: docs/src/main/asciidoc/_includes/devtools/create-app.adoc (hunk 3 above)
Comment (MichalMaler) [reply docs/src/main/asciidoc/_includes/devtools/create-app.adoc]: @gsmet Hello. The main purpose was to have it unified so that it is easier for me to search for these headers and some lines of code below them, and replace them with different lines of code. To have it automated, I needed to have it consistent first.

Such as, for example, if i would be doing this in IDEA:

**Regex:**
((\*) .*
\+)?
\[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"\]
\.pom.xml
----
((.*\n)+?)----
(\+?)
\[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"\]
\.build.gradle
----
((.*\n)+?)----


**Replacement:**
$1
$2* Using Maven:
+
--
[source,xml]
----
$3----
--
+
$2* Using Gradle:
$5
--
[source,gradle]
----
$6----
--


I am not a coder and not a regex expert, so I asked my friend @jmartisk if he could help me with composition of a bash script so that all writers can use it, not matter what IDE or Linux they use.
Problem is, the original regexs that works fine withing IDEa, doesnt work in bash, and some tweak and limitations needs to be addressed.
Also, my motivation here is not to put more burden on your shoulders @gsmet . I wanted to take this issue completely out of Quarkus , and using the script to patch the RHBQ files before each release, or when needed..

To make the coding easier, I want to reach some level of consistency at least in these files (which are completely fine and written in a super smart way, but i need to apply some cosmetics to workaround lacking and abandoned dowstream publication system) that are put in includes.

File: docs/src/main/asciidoc/_includes/devtools/create-app.adoc (hunk 4 above)
Comment (MichalMaler) [reply docs/src/main/asciidoc/_includes/devtools/create-app.adoc:57]: The inclide acts as a unit. If I want to glue that unit to something else, for example to a numbered procedural step, I need to use `+` in the empty lines between the step and the include. IF the include ends with an empty row, the `+` glue will be not effective and will not do the job, not aling the contet, and the numbering sequence will be broken. This mean that the next numbered step, which should be 2, will be one again. Thats how asciidoc works.

Here is one example (just for the demo):
image

image

I find out that we can work around that by wrapping the whole include in the `====` blocks. Then the empty line could be there, but if somebody would like to, for example, stick a NOTE to a snippet, and the note should not be the part of the snippet, we need to use `+` to align it properly. If the snipped will be ending with empty line and the whole thing around the `::include` will be not wrapped, it will not work.

Comment (MichalMaler) [reply docs/src/main/asciidoc/_includes/devtools/create-app.adoc:57]: Oh, my apologies. I thought you are pointing to the lines deleted at the end of the file. Yeah, I think I will put these back.

File: docs/src/main/asciidoc/_includes/devtools/create-cli.adoc (hunk 5 above)
Comment (MichalMaler) [reply docs/src/main/asciidoc/_includes/devtools/create-cli.adoc:55]: They are back now.

**Comment by MichalMaler** [comment]
@gsmet Your suggestions and concerns were addressed. The reason for the bottom empty line deletion is explained in the comments. Sentences and their style tweak according to the guide.
All headers now have this format:
```
[source,bash,subs=attributes+, role="secondary asciidoc-tabs-sync-gradle"]
```
Kindly asking for your review when you will have space for it. Thank you.

**Comment by MichalMaler** [comment]
Closing this and proposing the adjusted changes in https://github.com/quarkusio/quarkus/pull/35316
